ui/app/adapters/clients/activity.js

```diff
@@ -1,30 +1,56 @@
 import Application from '../application';
+import { zonedTimeToUtc } from 'date-fns-tz'; // https://github.com/marnusw/date-fns-tz#zonedtimetoutc
 
 export default Application.extend({
-  queryRecord(store, type, query) {
-    let url = `${this.buildURL()}/internal/counters/activity`;
-    // Query has startTime defined. The API will return the endTime if none is provided.
-    return this.ajax(url, 'GET', { data: query }).then((resp) => {
-      let response = resp || {};
-      // if the response is a 204 it has no request id (ARG TODO test that it returns a 204)
-      response.id = response.request_id || 'no-data';
-      return response;
-    });
-  },
-  // called from components
-  queryClientActivity(start_time, end_time) {
+  checkTimeType(query) {
+    let { start_time, end_time } = query;
     // do not query without start_time. Otherwise returns last year data, which is not reflective of billing data.
     if (start_time) {
-      let url = `${this.buildURL()}/internal/counters/activity`;
-      let queryParams = {};
-      if (!end_time) {
-        queryParams = { data: { start_time } };
+      // check if start_time is a RFC3339 timestamp and if not convert to one.
+      if (start_time.split(',').length > 1) {
+        let utcDate = this.utcDate(
+          new Date(Number(start_time.split(',')[1]), Number(start_time.split(',')[0] - 1))
+        );
+        start_time = utcDate.toISOString();
+      }
+      // look for end_time. If there is one check if it is a RFC3339 timestamp otherwise convert it.
+      if (end_time) {
+        let utcDateEnd = this.utcDate(
+          new Date(Number(end_time.split(',')[1]), Number(end_time.split(',')[0]))
+        );
+        // ARG TODO !!!! SUPER IMPORTANT, with endDate you need to make it the last day of the month, right now it's the first!!!
```

Comment:
Coming back to this in the next PR.

Reply:
Thank you for the method. We ended up approaching this another way after some discussion and confirmation with the backend. This is no longer an issue. Confirmed and tested.